### PR TITLE
Add validators to config

### LINKED ISSUE #200: Add validators for new config file
In #189 I already added a few simple validators, but I want to expand this to add more. Many of the variables have listed ranges here: https://fwl-proteus.readthedocs.io/en/latest/usage/

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR adds validation to the config file. I'm using https://github.com/FormingWorlds/PROTEUS/blob/main/docs/config_old.md, which nicely lists all the domains, as a basis.

Closes #200

### CODE REVIEW

**Review by nichollsh — CHANGES_REQUESTED** [review]
Looks good overall. I added a few comments for minor changes.

File: input/hd63433d.toml [hunk 1]
```diff
@@ -74,7 +74,7 @@ author  = "Harrison Nicholls, Tim Lichtenberg"
         # steady state
         [params.stop.steady]
             enabled  = true
-            F_crit   = 1.8        # Maximum absolute value of F_atm allowed for convergence
```
[the remark this answers is not in the document]
Comment (stefsmeets) [reply input/hd63433d.toml:77]: What should the range be then? Currently it is 0-1

[the remark this answers is not in the document]
Comment (nichollsh) [reply input/hd63433d.toml:77]: The valid range for `F_crit` and `dprel` is >0.

File: input/jgr_grid.toml [hunk 2]
```diff
@@ -219,7 +219,7 @@ author  = "Harrison Nicholls, Tim Lichtenberg"
         CH_ratio    = 1.0       # C/H ratio in mantle/atmosphere system
         H_oceans    = 6.0       # Hydrogen inventory in units of equivalent Earth oceans, by mass
         N_ppmw      = 2.0       # Nitrogen inventory in ppmw relative to mantle mass, by mass
-        S_ppmw      = 0.0       # Sulfur inventory in ppmw relative to mass of melt
```
[the remark this answers is not in the document]
Comment (nichollsh) [reply input/jgr_grid.toml:222]: @stefsmeets: note that the value of 300 ppmw is fine for the other configs. It's just this one that needs to be zero.

**Comment by stefsmeets** [comment]
Thanks @nichollsh , let me know if you are happy with the PR now!

**Review by nichollsh — APPROVED** [review]

**Pull request merged**